### ISSUE TITLE
Use `engines` field to avoid breaking changes.

### ISSUE THREAD

**Issue by WebReflection** [peerigon/updtr#45]
### Problem

I was testing `updtr` on a project running on node `0.12.x`. All tests were passing so I've tried to update from `eslintify` version `2.x` to `eslintify` version `3.x`.

Problem is, `eslintify` v3 is dependent of `eslint` which is compatible [node >= 4](https://github.com/eslint/eslint/blob/master/package.json#L118-L120) engines due usage of `const` and some other modern syntax but `eslintify` was part of a `browserify` task that wasn't involved with tests, only a watch while editing before building for prod.
### TL;DR

Running against tests doesn't necessarily mean all libraries/modules in the package list are used.
I understand `updtr` cannot do magic but at least it could compare current module and all its dependencies against current node engine version and avoid, where specified in the `package.json` engines fields, any sort of trouble (or warn about possible engine version conflict and ask confirmation for the upgrade)

Thanks for considering this.
